velox/expression/tests/ExpressionFuzzer.cpp

```diff
@@ -536,6 +538,39 @@ std::vector<core::TypedExprPtr> ExpressionFuzzer::generateRegexpReplaceArgs(
   return inputExpressions;
 }
 
+std::vector<core::TypedExprPtr> ExpressionFuzzer::generateSwitchArgs(
+    const CallableSignature& input) {
+  VELOX_CHECK_EQ(
+      input.args.size(),
+      2,
+      "Only two inputs are expected from the template signature.");
+  size_t cases = boost::random::uniform_int_distribution<uint32_t>(1, 5)(rng_);
+  bool useFinalElse =
+      boost::random::uniform_int_distribution<uint32_t>(0, 1)(rng_) > 0;
+
+  // Reduce the level of nesting to prevent the size of the expression tree to
+  // blow up significantly.
+  ScopedVarSetter nestingSetter(
+      &remainingLevelOfNesting_, std::min<int>(remainingLevelOfNesting_, 3));
```

Comment:
as I mentioned in the commit message, It reduces the nesting level when being used to avoid blowing up the expression tree as it can add up to 11 more branches to it. To give you an example of a blown up tree: if it gets selected at the top level with 9 remaining nesting levels, it can also show up as a subexpression in the tree and can result in > 500 columns being used at the end (if column re-use is disabled). Perhaps we can reduce it by an additional 1 instead of capping it at 3. All these are just empirical values so anything reasonable that we all agree on should work.

Reply:
I'm trying to understand your example. Let's say the max level of nesting is 3 and a switch expression is selected at the top level and have 2 cases (i.e., two conditions plus two then-clauses, but no else-clause). Then the max level of nesting of each condition or each then-clause will be 3 - 1 = 2. Would this make sense?

Reply:
generateSwitchArgs() is called after generateExpression() which has already reduced the count by 1, so setting the level to 3 here would mean that for each condition or then-clause the max level will be 3.
For context, I ran the fuzzer with 3 as the upper limit for a while and the max number of columns I saw was around 200